### ISSUE TITLE
require don't work inside template

### ISSUE THREAD

**Issue by aleeksunder** [jadejs/jade#1810]
Hi! Happy New Year =)
#### tl;dr
```jade
- var marked = require( 'marked' );
```
not working inside jade template with Jade v1.8.2 CLI on Windows Server 2012 R2 with x64 NodeJS v0.10.35. Not pretty sure what info to provide additionally, since I've found similar closed issue #1782. Really don't know maybe this fails due to os, platform, or CLI method.
Got the similar #1782 as response from CLI:
```
              throw e
                    ^
TypeError: test.jade:1
  > 1| - var marked = require( 'marked' )
    2| ul: li!= marked( '*Hello*' )
    3|

undefined is not a function
    at eval (eval at <anonymous> (R:\soft\NodeJS\x64\node_modules\jade\lib\index.js:190:8), <anonymous>:11:14)
    at eval (eval at <anonymous> (R:\soft\NodeJS\x64\node_modules\jade\lib\index.js:190:8), <anonymous>:25:22)
    at res (R:\soft\NodeJS\x64\node_modules\jade\lib\index.js:191:38)
    at R:\soft\NodeJS\x64\node_modules\jade\bin\jade.js:190:48
    at R:\soft\NodeJS\x64\node_modules\jade\node_modules\mkdirp\index.js:47:26
    at Object.oncomplete (fs.js:108:15)
```
If you're busy you really do not need to read the rest, because it is just like "pleeeeeeeeeease fix it for me, I will explain why I need so much" and that kind of stuff. =)))

#### The rest =)
Your tool is extremely useful for generating static html files, I use it for render my notes with a predefined template, and i'm very happy with:
```shell
jade --pretty note.jade
```
since I can put this command in association with .jade file and it just works with double-click.
So finally I want to have couple of files:
#### variable.jade
```jade
extend note.jade

block note-variable

    - var title = [ 'Title' , 'Additional Smaller Title' ];
    - var image = { exists : false , width : 100 , height : 100 , alt : 'Description' , src : '' };
    - var original = [ 'paragraph 1' , '_paragraph 2_' , 'paragraph *3*' ];
```
#### template.jade
```jade
extend page.jade
- var marked = require( 'marked' );

block note-variable

section.note
    block note-image
        if ( image.exists )
            img(
                src = '#{ image.src }'
                alt = '#{ image.alt }'
                width = '#{ image.width }'
                height = '#{ image.height }'
            )

    article.original

        block note-header
            header
                each value , key in title
                    #{ 'h' + ( key + 1 ) }
                        | #{ value }

        block note-original
            each paragraph in original
                != marked( paragraph )
```
and the `page.jade` which is a full html-document container.

Basically I just want to keep things as simple as it can be.
So to make a new note i can just edit variable.jade, double-click it and get an html document with all the stuff.

I know I can make another JavaScript file which will pass an `marked` object and render template with Jade API call... But...
I would not have created a issue, but it is so close to keep things DRY and simple with your tool...
The only two features which will make me happy are:
1. working `require`, so I can render markdown and maybe calculate image size with another NodeJS module
2. multiline code after dash, so I can keep my `variable.jade` more clear with line breaks.
 I've seen many discussions on how that may be implemented and just want to put my +1 for this method:
```jade
-
    var title = [
            'Title'
          , 'Additional Smaller Title'
        ]
      , image = {
            exists : false
          , width  : 100
          , height : 100
          , alt    : 'Description'
          , src    : ''
        }
      , original = [
            'paragraph 1'
          , '_paragraph 2_'
          , 'paragraph *3*'
      ]
    ;
```
P.S. And option to render end-files with 4 spaces indent will be very, very, very beautifull )))

**Comment by vendethiel** [comment]
You probably didn't pass `require` as a global or a local.

**Comment by aleeksunder** [comment]
How can I pass `require` as global or local with CLI jade call
`jade --pretty note.jade`
Can I pass it through this?

**Comment by aleeksunder** [comment]
Just tried some variations of
`jade --obj "{globals:['require']}" --pretty variable.jade`
and even
`jade --obj "{pretty:true}" variable.jade`
with different quotes and brackets and CLI output always was
```
  throw err;
        ^
RangeError: variable.jade:8
    6|     - var image = { exists : false , width : 100 , height : 100 , alt : 'Description' , src : '' };
    7|     - var original = [ 'paragraph 1' , '_paragraph 2_' , 'paragraph *3*' ];
  > 8|

Maximum call stack size exceeded
```
Am I missing something? Just don't understand how the CLI `--obj` option works

**Comment by aleeksunder** [comment]
And finally tried to put `variable.json` file near to `variable.jade`
#### variable.json
```
{
    "pretty": true
}
```
launched with this command:
```
jade --obj variable.json variable.jade
```
and the same CLI output:
```
  throw err;
        ^
RangeError: variable.jade:8
    6|     - var image = { exists : false , width : 100 , height : 100 , alt : 'Description' , src : '' };
    7|     - var original = [ 'paragraph 1' , '_paragraph 2_' , 'paragraph *3*' ];
  > 8|
```
and nothing seems to work... Is it a bug?

**Comment by aleeksunder** [comment]
Oh, guys, sorry for such a fail, just did't realize what exactly i'm doing wrong =)
here is the command line for those who may stucked with same problem:
```
jade --obj "{require:require}" --pretty variable.jade
```
I'll close the issue... Can you please give some comments on how can I pass Jade `options` object through the command line, for example if i want 4 spaces indent?
```
jade --obj "{pretty:'    '}" variable.jade
jade --obj "{options:{pretty:'    '}}" variable.jade
```
this is not working...
